src/lib/user-ip-address.js

```diff
@@ -0,0 +1,22 @@
+const validateIPaddress = require("./validate-ip-address");
+module.exports = (request) => {
+  const forwarded = request.headers["forwarded"];
+  if (forwarded) {
+    var ipAddressArray = forwarded.split(/[=;]/);
+    var clientIpAddress;
+    for (
+      var ipAddressCount = 0;
```

Comment:
Instead of iterating through the string in this method, is there not anyway we can build an object from the string and take the value associated from the `for` key.

For example -
```javascript
const myObj = {};
const forwardedArray = forwarded.split(";");
for(const val of forwardedArray) {
   const [key, value] = val.split("=");
   myObj[key] = value;
} // this for look should create an object from that forwarded string.

const clientIp = myObj.for;
return validateIPaddress(clientIp) ? clientIp : null;

```

Reply:
I think we have confidence it will always be the `for` value. But the location in the string is up for debate.

Reply:
One idea I had yesterday was seeing if the express library will automatically parse the string into a header object 🤔